conjure-codegen/src/example_types/product/covariant_optional_example.rs

```diff
@@ -2,68 +2,23 @@ use conjure_object::serde::{ser, de};
 use conjure_object::serde::ser::SerializeStruct as SerializeStruct_;
 use std::fmt;
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[conjure_object::private::staged_builder::staged_builder]
+#[builder(crate = conjure_object::private::staged_builder, update, inline)]
 pub struct CovariantOptionalExample {
+    #[builder(default, into)]
     item: Option<conjure_object::Any>,
```

Comment:
Why can it just do `into` here, whereas the Any type in a map or vec needs a customer serializer like https://github.com/palantir/conjure-rust/pull/349/files#diff-8fcc3688b43688b6b278cb5ba8f57cc9c0446f2b1200d948142255818816227cR11-R17?

Reply:
As an aside, could we just define the custom serializer once somewhere so it doesn't have to keep redefining it?

Reply:
I actually think this is specific to the `Option`, because `conjure_object::Any` has the custom serializer here https://github.com/palantir/conjure-rust/pull/349/files#diff-450a3ab87b025c7854669c7345e9221165642460bc03581346e2ed08db393726R11-R18

Reply:
We can use `into` for places where we want the conversion logic to just use an `Into` implementation.

You can see in the deleted manual builder code below that the behavior here matches what we were doing previously.